contracts/RockPaperScissors.sol

```diff
@@ -0,0 +1,286 @@
+    }
+
+    event PlayerOneMoves(
+        address indexed player,
+        bytes32 indexed gameKey,
+        uint256 value
+    );
+
+    event PlayerTwoMoves(
+        address indexed player,
+        bytes32 indexed gameKey,
+        uint256 value
+    );
+
+    event PlayerTwoReveals(
+        address indexed player,
+        bytes32 indexed gameKey,
+        uint8 move
+    );
+
+    event WinnerRewarded(
+        address indexed player,
+        bytes32 indexed gameKey,
+        uint256 amount
+    );
+
+    event  ForfeitPaid(
+        address indexed player,
+        bytes32 indexed gameKey,
+        uint256 amount
+    );
+
+    event WithDraw(
+        address indexed withdrawer,
+        uint amount
+    );
+
+    // The creator's move is embedded in the game key
+    function playerOneMove(bytes32 gameKey, uint stake) external payable whenNotPaused {
+
+        require(gameKey != bytes32(0), "Game key cannot be empty");
+        require(stake > 0, "Stake cannot be zero");
+
+        Game storage game = games[gameKey];
+        require(game.step == Steps.INIT, INVALID_STEP_MSG);
+
+        game.playerOne = msg.sender;
+        game.stake = stake;
+        game.step = Steps.PLAYER_ONE_MOVE;
+        depositStake(stake);
+
+        emit PlayerOneMoves(msg.sender, gameKey, stake);
+    }
+
+    /*
+    * The opponent joins the game
+    */
+    function playerTwoMove(bytes32 gameKey, bytes32 moveHash) external payable whenNotPaused {
+
+        Game storage game = games[gameKey];
+        uint256 stake = game.stake;
+        require(stake != 0, GAME_NOT_FOUND_MSG);
+        require(game.step == Steps.PLAYER_ONE_MOVE, INVALID_STEP_MSG);
+
+        game.playerTwo = msg.sender;
+        game.playerTwoMoveHash = moveHash;
+        game.step = steps.PLAYER_TWO_MOVE;
+        depositStake(stake);
+
+        emit PlayerTwoMoves(msg.sender, msg.value, gameKey);
+    }
+
+    function depositStake(uint256 stake) internal {
+        uint256 senderBalance = balances[msg.sender];
+
+        if (msg.value == stake) {
+            return;
+        }
+
+        if (msg.value < stake) {
+            require(senderBalance >= stake.sub(msg.value), "Insufficient balance");
+            balances[msg.sender] = senderBalance.sub(stake.sub(msg.value));
+        } else {
+            balances[msg.sender] = senderBalance.add(msg.value.sub(stake));
+        }
+
+        Deposit(msg.sender, msg.value);
+    }
+
+    function playerOneReveal(bytes32 gameKey, bytes32 secret, uint8 move) external whenNotPaused {
+
+        require(move < 3, INVALID_MOVE_MSG);
+        Game storage game = games[gameKey];
+        uint256 stake = game.stake;
+        require(stake != 0, GAME_NOT_FOUND_MSG);
+        require(game.step == Steps.PLAYER_TWO_MOVE, INVALID_STEP_MSG);
+        address playerOne = game.playerOne;
+        require(msg.sender == playerOne, ONLY_PLAYER_ONE_CALL_MSG);
+
+
+        //Validate move
+        bytes32 expectedMoveHash = createGameKey(secret, move);
```

Comment:
Having game key in the function name and hash in the result is a bit confusing.

Reply:
I was also thing it might be confusing. I will revisit.